libs/parallelism/execution/include/hpx/execution/algorithms/detach.hpp

```diff
@@ -0,0 +1,115 @@
+//  Copyright (c) 2021 ETH Zurich
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#pragma once
+
+#include <hpx/config.hpp>
+#include <hpx/allocator_support/allocator_deleter.hpp>
+#include <hpx/allocator_support/internal_allocator.hpp>
+#include <hpx/execution/algorithms/detail/partial_algorithm.hpp>
+#include <hpx/execution_base/operation_state.hpp>
+#include <hpx/execution_base/sender.hpp>
+#include <hpx/functional/invoke_result.hpp>
+#include <hpx/functional/tag_fallback_invoke.hpp>
+#include <hpx/modules/memory.hpp>
+
+#include <atomic>
+#include <cstddef>
+#include <exception>
+#include <memory>
+#include <tuple>
+#include <type_traits>
+#include <utility>
+
+namespace hpx { namespace execution { namespace experimental {
+    namespace detail {
+        template <typename S>
+        struct operation_state_holder
+        {
+            struct detach_receiver
+            {
+                hpx::intrusive_ptr<operation_state_holder> os;
+
+                template <typename E>
+                void set_error(E&&) noexcept
+                {
+                    std::terminate();
+                }
+
+                void set_done() noexcept
+                {
+                    os.reset();
+                };
+
+                template <typename... Ts>
+                void set_value(Ts&&...) noexcept
+                {
+                    os.reset();
+                }
+            };
+
+            std::atomic<std::size_t> count{0};
```

Comment:
This should be slightly less heavyweight:
```suggestion
            hpx::util::atomic_count count{0};
```

Reply:
You might want to consider making the `count` private to avoid anybody starts messing with it. This would also require for the `intrusive_ptr_addref/release` functions to become friends of this class.

Reply:
Note: I'll apply this in #5274. Is the main difference that `atomic_count` uses more relaxed memory orderings by default?